File: src/main/java/com/codecool/repvault/application/DTOs/outgoing/SessionResponseDTO.java
package com.codecool.repvault.application.DTOs.outgoing;

import com.codecool.repvault.domain.entities.Session;
import lombok.Getter;

import java.time.ZonedDateTime;
import java.util.List;

@Getter
public class SessionResponseDTO {
    private Long id;
    private String name;
    private String description;
    private List<ExerciseResponseDTO> exercises;
    private ZonedDateTime start;
    private ZonedDateTime end;

    public SessionResponseDTO(Session session) {
        this.id = session.getId();
        this.description = session.getDescription();
        this.name = session.getName();
        this.exercises = session.getExercises().stream()
                .map(ExerciseResponseDTO::new)
                .toList();
        this.start = session.getStart();
        this.end = session.getEnd();
    }
}

File: src/main/java/com/codecool/repvault/application/DTOs/outgoing/SessionResponseDTO.kt
package com.codecool.repvault.application.DTOs.outgoing

import com.codecool.repvault.domain.entities.Exercise
import com.codecool.repvault.domain.entities.Session
import java.time.ZonedDateTime

class SessionResponseDTO(session: Session) {
    val id: Long = session.id!!
    val user: UserResponseDTO = UserResponseDTO(session.user!!)
    val volume: Double? = session.volume
    val name: String = session.name!!
    val description: String? = session.description
    val exercises: MutableList<ExerciseResponseDTO> = session.exercises.stream()
        .map<ExerciseResponseDTO> { exercise: Exercise -> ExerciseResponseDTO(exercise) }
        .toList()
    val start: ZonedDateTime = session.start!!
    val end: ZonedDateTime = session.end!!
}
